src/components/VmDetails/cards/DisksCard/DiskImageEditor.js

```diff
@@ -155,47 +148,59 @@ class DiskImageEditor extends Component {
     this.setState({ showModal: false })
   }
 
-  // NOTE: Add and edit both can use the same composeDisk() since the add and edit
-  //       sagas will use what they need from the composed disk.  The sagas ultimately
-  //       control the REST calls and data.
   composeDisk () {
-    const { vm, disk } = this.props
-    const vmDiskInSameStorageDomain =
-      vm.get('disks') &&
-      vm.get('disks').find(disk => disk.get('storageDomainId') === this.state.values.storageDomain)
-
-    const iface =
-      (disk && disk.get('iface')) ||
-      (vmDiskInSameStorageDomain && vmDiskInSameStorageDomain.get('iface')) ||
-      'virtio_scsi'
-
-    const provisionedSize = disk
-      ? disk.get('provisionedSize') + this.state.values.size * 1024 ** 3
-      : this.state.values.size * 1024 ** 3
-
-    const bootable = this.state.values.bootable
-
-    const newDisk = {
-      ...DISK_DEFAULTS,
-      attachmentId: disk && disk.get('attachmentId'),
-      storageDomainId: this.state.values.storageDomain,
-
-      iface,
-      id: this.state.id,
-      name: this.state.values.alias,
-      provisionedSize,
-      bootable,
-
-      // the __diskType__ field maps to format + sparse REST Disk attributes
-      format: 'raw', // Match webadmin behavior, disks are created as 'raw'
-      sparse: this.state.values.diskType === 'thin',
-    }
+    const { vm, disk, storageDomains } = this.props
+    const { values } = this.state
+
+    if (disk) { // edit
+      const provisionedSize = disk.get('provisionedSize') + values.size * 1024 ** 3
+
+      return disk.get('type') === 'image'
+        ? { // image disk (change name, size, bootable)
+          attachmentId: disk.get('attachmentId'),
+          id: disk.get('id'),
+
+          bootable: values.bootable,
+          name: values.alias,
+          provisionedSize,
+        }
+        : { // cinder or lun disk (only change name and bootable)
+          attachmentId: disk.get('attachmentId'),
+          id: disk.get('id'),
+          type: disk.get('type'),
+
+          bootable: values.bootable,
+          name: values.alias,
+        }
+    } else { // new
+      const vmDiskInSameStorageDomain =
+        vm.get('disks') &&
+        vm.get('disks').find(disk => disk.get('storageDomainId') === values.storageDomain)
+
+      const iface = vmDiskInSameStorageDomain
```

Comment:
>   1. `iface` is marked as optional  in disk - we should probably check if it exists

It is part of the disk attachment, not the disk itself, and is always received from rest.  The internal data type has it marked optional since it is not require for all disk/attachment edits.

> 
>   2. we have some heuristic here based on existing disks - why we cannot (directly) get  the `iface` supported by selected storage domain ?

That's simple.  `iface` is part of the VM definition, not the storage domain.  Storage domains don't care what qemu interface is used to present the disk to a VM.  I believe the `iface` default in webadmin is `virtio_scsi`.  I don't actually remember why the selection is like this.  Probably to support adding disks to VMs that can't use `virtio_scsi` without having to expose "non-admin users" to minutia like VM disk attachment interface.

Reply:
> > 1. `iface` is marked as optional  in disk - we should probably check if it exists
> 
> It is part of the disk attachment, not the disk itself, and is always received from rest. The internal data type has it marked optional since it is not require for all disk/attachment edits.

OK. So it's another case of read vs update conflict in REST API type definitions. All of them require a common solution.

> > 2. we have some heuristic here based on existing disks - why we cannot (directly) get the iface supported by selected storage domain ?
>
>That's simple.

OK. Please describe it as a comment in the code.